src/main/java/net/dancier/chatdancer/controllers/ChatController.java

```diff
@@ -0,0 +1,118 @@
+package net.dancier.chatdancer.controllers;
+
+import lombok.RequiredArgsConstructor;
+import net.dancier.chatdancer.dtos.*;
+import net.dancier.chatdancer.models.Chat;
+import net.dancier.chatdancer.models.Message;
+import net.dancier.chatdancer.services.ChatService;
+import net.dancier.chatdancer.utils.BadRequestException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.UUID;
+
+@RestController()
+@RequestMapping("/chats")
+@RequiredArgsConstructor
+public class ChatController {
+
+    public static final Logger log = LoggerFactory.getLogger(ChatController.class);
+    private final ChatService chatService;
+
+    @GetMapping
+    public ResponseEntity<ChatsResponseDto> getAllChats(@RequestParam UUID dancerId) {
+
+        List<Chat> allChatsForDancer = chatService.getAllChatsForDancer(dancerId);
+        List<ChatResponseDto> chatResponseDtoList = allChatsForDancer.stream().map(this::convertChatToDto).toList();
+        ChatsResponseDto chatsResponseDto = ChatsResponseDto.builder().chats(chatResponseDtoList).build();
+
+        return new ResponseEntity<>(chatsResponseDto, HttpStatus.OK);
+    }
+
+    @GetMapping("/{chatId}")
+    public ResponseEntity<ChatResponseDto> getChatById(@PathVariable UUID chatId) {
+
+        Chat requestedChat = chatService.getChatById(chatId);
+
+        return new ResponseEntity<>(convertChatToDto(requestedChat), HttpStatus.OK);
+    }
+
+    @GetMapping("/{chatId}/messages")
+    public ResponseEntity<MessagesResponseDto> getAllMessagesForChat(@PathVariable UUID chatId) {
```

Comment:
here you also need to integrate the query parameters mentioned in the swagger file to define from which message on messages should be returned. But this can be part of a follow up ticket